homeassistant/components/panasonic_viera/__init__.py

```diff
@@ -252,14 +246,18 @@ async def async_get_device_info(self):
     async def _handle_errors(self, func, *args):
         """Handle errors from func, set available and reconnect if needed."""
         try:
-            return await self._hass.async_add_executor_job(func, *args)
+            result = await self._hass.async_add_executor_job(func, *args)
+            self.state = STATE_ON
+            self.available = True
+            return result
         except EncryptionRequired:
             _LOGGER.error(
                 "The connection couldn't be encrypted. Please reconfigure your TV"
             )
+            self.available = False
         except (TimeoutError, URLError, SOAPError, OSError):
             self.state = STATE_OFF
-            self.available = self._on_action is not None
+            self.available = True
```

Comment:
This seems incorrect?
How can the device be available when there are communication issues?

Reply:
It seems incorrect but unfortunately, at leat in my case, it is not. There will occur SOAP errors as soon as the device is in standby, even if you are able to turn it on via SOAP successfully, every call to get an info from the device (eg. Volume) will result in an error. I guess this is the main reason why the TV is 'unavailable' all the time for some users in the current version. 

But I think I need to specify more clearly in which case this is really an error and in which it is a 'normal/available' behaviour. But this is kinda hard for me since I just know the behaviour of one model...

Reply:
Hope it helps :) And yes @frenck its incorrect in the underlying python lib... as far as I see.

Ref:
https://github.com/home-assistant/core/issues/34894#issuecomment-735249190

Reply:
I have a TH-55AX670Z TV and that also has SOAP errors when in standby. When its not in standby I still get the occasional timeout, even with an ethernet cable instead of wifi. It makes it hard to tell if the TV is on or off before toggling the power. It was so unreliable I've ended up modifying this code in a custom component, to check if its on by sending a few get_volume requests before toggling the power. Now finally telling it to turn off wont turn it back on.